hs_core/views/resource_folder_hierarchy.py

```diff
@@ -132,25 +127,33 @@ def data_store_structure(request):
         logical_file_type = ''
         logical_file_id = ''
         aggregation_name = ''
-        is_single_file_aggregation = ''
-        if resource.resource_type == "CompositeResource":
-            if f.has_logical_file:
-                logical_file_type = f.logical_file_type_name
-                logical_file_id = f.logical_file.id
-                aggregation_name = f.aggregation_display_name
-                is_single_file_aggregation = f.logical_file.is_single_file_aggregation
-                if 'url' in f.logical_file.extra_data:
-                    f_ref_url = f.logical_file.extra_data['url']
+        if f.has_logical_file:
+            main_extension = f.logical_file.get_main_file_type()
+            if not main_extension:
+                # accept any extension
+                main_extension = ""
```

Comment:
line 132 to 134 will not be needed if get_main_file_type() just returns empty string for unrecognized file type extensions.

Reply:
it returns None and I don't want to update it in case code elsewhere expects None.